engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/HistoricProcessInstanceManager.java

```diff
@@ -241,6 +241,9 @@ public void addRemovalTimeById(String processInstanceId, Date removalTime) {
     commandContext.getByteArrayManager()
       .addRemovalTimeToByteArraysByProcessInstanceId(processInstanceId, removalTime);
 
+    commandContext.getAuthorizationManager()
+        .addRemovalTimeToAuthorizationsByProcessInstanceId(processInstanceId, removalTime);
```

Comment:
Should we skip this entirely if `enableHistoricInstancePermissions` is disabled?

Reply:
👍 This makes might increase performance and makes the behavior consistent with how we do it when DMN is disabled.